proto/aserto/directory/reader/v2/reader.proto

```diff
@@ -4,36 +4,366 @@ package aserto.directory.reader.v2;
+            post: "/api/v2/directory/relation_types"
+            body: "*"
+        };
+        option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+            tags: "directory"
+            summary: "Get relation types"
+            description: "Returns relation types, optionally by object_type."
+            operation_id: "directory.relation_types.get"
+            deprecated: false
+            security: {
+                security_requirement: {
+                    key: "TenantID";
+                    value: {}
+                }
+                security_requirement: {
+                    key: "DirectoryAPIKey";
+                    value: {}
+                }
+            }
+        };        
+    };
     
     // permission metadata methods
-    rpc GetPermission(GetPermissionRequest) returns (GetPermissionResponse) {};
-    rpc GetPermissions(GetPermissionsRequest) returns (GetPermissionsResponse) {};
+    rpc GetPermission(GetPermissionRequest) returns (GetPermissionResponse) {
+        option (google.api.http) = {
+            post: "/api/v2/directory/permission"
+            body: "*"
+        };
+        option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+            tags: "directory"
+            summary: "Get permission"
+            description: "Returns permission by name."
+            operation_id: "directory.permission.get"
+            deprecated: false
+            security: {
+                security_requirement: {
+                    key: "TenantID";
+                    value: {}
+                }
+                security_requirement: {
+                    key: "DirectoryAPIKey";
+                    value: {}
+                }
+            }
+        };    
+    };
+
+    rpc GetPermissions(GetPermissionsRequest) returns (GetPermissionsResponse) {
+        option (google.api.http) = {
+            post: "/api/v2/directory/permissions"
+            body: "*"
+        };
+        option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+            tags: "directory"
+            summary: "Get permissions"
+            description: "Returns permissions."
+            operation_id: "directory.permissions.get"
+            deprecated: false
+            security: {
+                security_requirement: {
+                    key: "TenantID";
+                    value: {}
+                }
+                security_requirement: {
+                    key: "DirectoryAPIKey";
+                    value: {}
+                }
+            }
+        };    
+    };
     
     // object methods
-    rpc GetObject(GetObjectRequest) returns (GetObjectResponse) {};
+    rpc GetObject(GetObjectRequest) returns (GetObjectResponse) {
+        option (google.api.http) = {
+            post: "/api/v2/directory/object"
+            body: "*"
+        };
+        option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
```

Comment:
The current usage of optional fields inside ObjectIdentifier prohibits the usage of them as path parameters

```
        option (google.api.http) = {
            post: "/api/v2/directory/object"
            body: "*"
            additional_bindings: [
            {
                get: "/api/v2/directory/object/{param.type}/{param.key}"
            }
            ]
        };
```

will result in 
```
>>> executing buf generate ../pb-directory/bin/directory.bin
Failure: plugin grpc-gateway: optional field not allowed in field path: type in param.type
Error: running "/Users/gertd/workspace/src/github.com/aserto-dev/go-directory/.ext/gobin/buf-v1.9.0/buf generate ../pb-directory/bin/directory.bin" failed with exit code 1
```